components/overlay_trigger.tsx

```diff
@@ -9,7 +9,11 @@ import {Provider} from 'react-redux';
 
 import store from 'stores/redux_store.jsx';
 
-export {BaseOverlayTrigger};
+interface HideableOverlayTrigger extends BaseOverlayTrigger {
+    hide: () => void
```

Comment:
Is there a way to just add the `hide` method to the `BaseOverlayTrigger` interface instead of defining something new? It would be nice to avoid the new name since people using this code don't really care about using the React Bootstrap interface or our custom one.

The reason we're hitting that though is because the type definitions are probably for a newer version of React Bootstrap than the outdated one we're stuck on currently.

Reply:
I have tried to work-around this one a little and, eventually, this was the only approach that sounded decent.
The issue is that the current typing definition for `react-bootstrap` does not expose the `hide()` method. I can try to see if there is any Typescript trip to _patch_ that definition and add the method there.

Reply:
@lmammino Have you tried loading the `react-bootstrap` types for an older version? I'm not sure if they exist and I seem to recall looking for them at one point, but if they're not officially provided it's possible that DefinitelyTyped has something provided.

Reply:
I'd assume they don't exist that far back. We're very far behind on them. What if you just import the provided type from them with a different name and call our version `BaseOverlayTrigger`?
```typescript
import {OverlayTrigger as OverlayTriggerInterface} from 'react-bootstrap';

interface BaseOverlayTrigger extends OverlayTriggerInterface {
    // same as before
}
```
I can't actually remember the reason that we export BaseOverlayTrigger separately from OverlayTrigger, but I'm just trying to avoid adding a third thing that's exported with all of them being the same as each other.

Reply:
I ended up doing something like this:

```typescript
import {OverlayTrigger as OriginalOverlayTrigger, OverlayTriggerProps} from 'react-bootstrap';
// ...
export type BaseOverlayTrigger = OriginalOverlayTrigger & {
    hide: () => void
}
```

What do you think?

Reply:
Yep, that works great!